### PR TITLE
Refactors & Enables species-sensor nanites

### PR DESCRIPTION
**Pull request by Kapu1178**
## About The Pull Request
Title

## Why It's Good For The Game
The potential is limitless. Felinids forcefully wagging tails (not lizards you guys are epic), detonating every non-felinid on the station, forcing ONLY felinids to nya. This is comedy gold waiting to happen. Why wasnt this enabled already?

Also, you can make oozelings not fucking DIE from blood purification

## Changelog
:cl:
add: species sensor nanites are functional and in Harmonic
/:cl:

### CODE REVIEW

**Comment by Sinestia** [comment]
Kapu the saviour of golden

**Comment by TeomanTheGreat** [comment]
Detonating every felinid on the station. Yes.

**Comment by heepox** [comment]
based. more ways to bully felinids.

**Comment by ghost** [comment]
Based as fuck

**Comment by ItsTisme** [comment]
But this will cause felined genocide and general racism!!


Besure to ping maintainers for speed merge this is awsome

File: code/modules/research/nanites/nanite_programs/sensor.dm [hunk 1]
```diff
@@ -265,27 +265,27 @@
...
-		"Pod" = /datum/species/pod,
-		"Fly" = /datum/species/fly,
-		"Felinid" = /datum/species/human/felinid,
-		"Jelly" = /datum/species/jelly
-	)
+	var/list/static/allowed_species
+
+/datum/nanite_program/sensor/species/New()
+    if(!length(allowed_species))
+        allowed_species = list()
```
Comment (utilbird) [inline code/modules/research/nanites/nanite_programs/sensor.dm]:
```suggestion
	var/list/static/allowed_species = list()

/datum/nanite_program/sensor/species/New()
    if(!length(allowed_species))
```

File: code/modules/research/nanites/rules.dm [hunk 2]
```diff
@@ -149,3 +149,43 @@
...
+	if(species_match_rule)
+		if(mode_rule)
+			return TRUE
+		else
+			return FALSE
+	else
+		if(!mode_rule)
+			return TRUE
+		else
+			return FALSE
```
Comment (utilbird) [inline code/modules/research/nanites/rules.dm]:
```suggestion
	return species_match_rule ? mode_rule : !mode_rule
```

**Review by Kapu1178 — COMMENTED** [review]
Im not sure this works, actually. This doesnt handle the rule being set to IS NOT, no?

**Review by utilbird — APPROVED** [review]
it works

**Pull request merged**